src/node/internal/events.ts

```diff
@@ -495,26 +497,34 @@ function _addListener(
           `added to an EventEmitter. Use ` +
           'emitter.setMaxListeners() to increase limit'
       );
-      // TODO(soon): Implement process.emitWarning and inspect
-      // // No error code for this since it is a Warning
-      // // eslint-disable-next-line no-restricted-syntax
-      // const w = new Error(
-      //   "Possible EventEmitter memory leak detected. " +
-      //     `${existing.length} ${String(type)} listeners ` +
-      //     `added to ${inspect(target, { depth: -1 })}. Use ` +
-      //     "emitter.setMaxListeners() to increase limit",
-      // );
-      // w.name = "MaxListenersExceededWarning";
-      // w.emitter = target;
-      // w.type = type;
-      // w.count = existing.length;
-      // process.emitWarning(w);
+      const w: EventEmitterError = Object.assign(
+        new Error(
+          'Possible EventEmitter memory leak detected. ' +
+            `${existing.length} ${String(type)} listeners ` +
+            `added to ${inspect(target, { depth: -1 })}. Use ` +
+            'emitter.setMaxListeners() to increase limit'
+        ),
+        {
+          name: 'MaxListenersExceededWarning',
+          emitter: target,
+          type: type,
+          count: existing.length,
+        }
+      );
+      process.emitWarning?.(w);
```

Comment:
Can you put this warning generation into a if statement and check if process.emitWarning is enabled or not? And can you also add a comment stating how and under which conditions process.emitWarning can be defined/undefined?

Reply:
I've added a comment clarifying this behaviour.